src/main/java/com/cloudant/client/api/ClientBuilder.java

```diff
@@ -0,0 +1,292 @@
+                    .port(port)
+                    .build().toURL();
+        } catch (MalformedURLException e) {
+            throw new CouchDbException(e);
+        }
+    }
+
+    public CloudantClient build() {
+
+        //Build properties and couchdb client
+        CouchDbProperties props = new CouchDbProperties(url.getProtocol(),
+                url.getHost(), url.getPort());
+
+        //Create cookie interceptor
+        if (this.username != null && this.password != null) {
+            //make interceptor if both username and password are not null
+
+            //Create cookie interceptor and set in HttpConnection interceptors
+            CookieInterceptor cookieInterceptor = new CookieInterceptor(username, password);
+
+            props.addRequestInterceptors(cookieInterceptor);
+            props.addResponseInterceptors(cookieInterceptor);
+        } else {
+            //If username or password is null, throw an exception
+            if (username != null || password != null) {
+                //Username and password both have to contain values
+                throw new CouchDbException("Either a username and password must be provided, or " +
+                        "both values must be null. Please check the credentials and try again.");
+            }
+        }
+
+
+        //If setter methods for read and connection timeout are not called, default values are used.
+        props.addRequestInterceptors(new TimeoutCustomizationInterceptor(connectionTimeout,
+                readTimeout));
+
+        //Set connect options
+        props.setMaxConnections(maxConnections);
+        props.setProxyURL(proxyURL);
+        if(proxyUser != null) {
+            //if there was proxy auth information create an interceptor for it
+            props.addRequestInterceptors(new ProxyAuthInterceptor(proxyUser,
+                    proxyPassword));
+        }
+        if (isSSLAuthenticationDisabled) {
+            props.addRequestInterceptors(SSLCustomizerInterceptor
+                    .SSL_AUTH_DISABLED_INTERCEPTOR);
+        } else {
+            if (authenticatedModeSSLSocketFactory != null) {
+                props.addRequestInterceptors(new SSLCustomizerInterceptor(
+                        authenticatedModeSSLSocketFactory
+                ));
+            }
+        }
+
+        //Set http connection interceptors
+        if(requestInterceptors != null) {
+            for(HttpConnectionRequestInterceptor requestInterceptor : requestInterceptors) {
+                props.addRequestInterceptors(requestInterceptor);
+            }
+        }
+        if(responseInterceptors != null) {
+            for(HttpConnectionResponseInterceptor responseInterceptor : responseInterceptors) {
+                props.addResponseInterceptors(responseInterceptor);
+            }
+        }
+
+        return new CloudantClient(props, gsonBuilder);
+    }
+
+    public ClientBuilder username(String username) {
```

Comment:
Should this be `setUsername` to be consistent with the other fields, or should the other fields lose their `set` prefixes?